apps/test/unit/lib/levelbuilder/data-docs-editor/DataDocFormEditorTest.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import {expect} from '../../../../util/reconfiguredChai';
+import DataDocFormEditor from '@cdo/apps/lib/levelbuilder/data-docs-editor/DataDocFormEditor';
+import {isolateComponent} from 'isolate-react';
+
+describe('DataDocFormEditor', () => {
+  let defaultProps;
+  const docKey = 'doc_key';
+  const docName = 'Name of Doc';
+  const docContent = 'Content of Doc';
+
+  beforeEach(() => {
+    defaultProps = {
+      dataDocKey: docKey,
+      originalDataDocName: docName,
+      originalDataDocContent: docContent
+    };
+  });
+
+  it('renders editor with props already loaded into entry fields', () => {
+    const wrapper = isolateComponent(<DataDocFormEditor {...defaultProps} />);
+    expect(wrapper.findAll('input')[0].props.value).to.equal(docKey);
+    expect(wrapper.findAll('input')[1].props.value).to.equal(docName);
+    expect(
+      wrapper.findOne('TextareaWithMarkdownPreview').props.markdown
+    ).to.equal(docContent);
+  });
+
+  it('modifies doc name', () => {
+    const wrapper = isolateComponent(<DataDocFormEditor {...defaultProps} />);
+    const newDocName = 'New Name of Doc';
+    wrapper.findAll('input')[1].props.onChange({target: {value: newDocName}});
+
+    expect(wrapper.findAll('input')[0].props.value).to.equal(docKey);
+    expect(wrapper.findAll('input')[1].props.value).to.equal(newDocName);
+    expect(
+      wrapper.findOne('TextareaWithMarkdownPreview').props.markdown
+    ).to.equal(docContent);
+  });
+
+  it('modifies doc content', () => {
+    const wrapper = isolateComponent(<DataDocFormEditor {...defaultProps} />);
+    const newDocContent = 'New Content of Doc';
+    wrapper
+      .findOne('TextareaWithMarkdownPreview')
+      .props.handleMarkdownChange({target: {value: newDocContent}});
+
+    expect(wrapper.findAll('input')[0].props.value).to.equal(docKey);
+    expect(wrapper.findAll('input')[1].props.value).to.equal(docName);
+    expect(
+      wrapper.findOne('TextareaWithMarkdownPreview').props.markdown
+    ).to.equal(newDocContent);
+  });
```

Comment:
it would be great to add an integration-style test here that covers:
1. pushing the save button
2. stubbing out the ajax request with sinon.fakeServer
3. stubbing out navigateToHref and verifying that it gets called properly

Reply:
Will do!